ext/-test-/string/fstring.c

```diff
@@ -8,8 +9,22 @@ bug_s_fstring(VALUE self, VALUE str)
     return rb_fstring(str);
 }
 
+VALUE
+bug_s_rb_enc_interned_str(VALUE self, VALUE encoding)
+{
+    return rb_enc_interned_str("foo", 3, RDATA(encoding)->data);
```

Comment:
Doesn't `rb_to_encoding(encoding)` feel cleaner?

Reply:
it is cleaner, but it does call `rb_encoding_check` which loads the encoding so prevent the bug from happening.